docs/reference/changelog.rst

```diff
@@ -22,11 +22,24 @@ Changelog
 2.20.1 (2025-MM-DD)
 -------------------
 
+New features:
+
+- Previously, when the Maven Use plugin updated ``pom.xml`` for self-contained projects,
+  it wouldn't reliably find the correct dependency versions on the host. It could
+  unpredictably declare the wrong package version, or select a vastly different version
+  despite a similar one being available.
+
+  With Craft Parts 2.20.1, the plugin now deterministically detects and matches the
+  dependency versions available on the host, aligning with how Maven normally behaves.
+  This change makes the plugin a drop-in replacement of Maven, for private networks.
+
 Bug fixes:
 
 - `#1007 <https://github.com/canonical/craft-parts/issues/1007>`_ When wild cards
   were used in an ``organize`` source path, an error would occur if files mapped to
   themselves. These cases are now ignored.
+- When rebuilding a part with the ``self-contained`` build attribute, the maven-use
+  plugin now guarantees to select the same dependency version.
```

Comment:
I honestly think we could skip this, since it's covered in the new features section.